lightning/src/sign/mod.rs

```diff
@@ -1888,6 +1874,35 @@ impl PhantomKeysManager {
 	}
 }
 
+/// An implementation of [`EntropySource`] using [`ChaCha20`].
+#[derive(Debug)]
+struct RandomBytes {
+	/// Seed from which all randomness produced is derived from.
+	seed: [u8; 32],
+	/// Tracks the number of times we've produced randomness to ensure we don't return the same
+	/// bytes twice.
+	index: AtomicCounter,
```

Comment:
> Isn't it unsafe if this isn't persisted and we will reset it to new counter? Do we expect uniq seed every time?

Yeah, a unique seed is expected if you don't want to produce the same values.

> Looking at this struct independently, it seems to imply "give me a static seed and it will work"

Depends on what you mean by "works". There isn't anything in the public docs claiming that restarting with the same seed will not produce the same values. Happy to reword the private docs here. I was just copying them.